sys/net/application_layer/dns/cache.c

```diff
@@ -137,10 +140,12 @@ int dns_cache_query(const char *domain_name, void *addr_out, int family)
         if (cache[i].hash == hash && (!addr_len || addr_len == _get_len(i))) {
             DEBUG("dns_cache[%u] hit\n", i);
             memcpy(addr_out, &cache[i].addr, _get_len(i));
+            mutex_unlock(&cache_mutex);
             return _get_len(i);
```

Comment:
I smell a potential race condition

Reply:
Fixed with the suggestion provided in https://github.com/RIOT-OS/RIOT/pull/18330#discussion_r924690579